### ISSUE TITLE
resource mongodbatlas_cluster

### ISSUE THREAD

**Issue by OlehHarmash** [mongodb/terraform-provider-mongodbatlas#1370]
Terraform v1.1.9
on windows_amd64
+ provider registry.terraform.io/mongodb/mongodbatlas v1.11.0

I have tried to add labels(TAGs) for cluster.
It shows changes in plan.
      - labels {
          - key   = "tf_key" -> null
          - value = "tf_value" -> null
        }
      + labels {
          + key   = "tf_key1"
          + value = "tf_value1"
        }
After applying I can see no tags in https://cloud.mongodb.com/v2/${MY_CLUSTTER_ID}#/clusters

**Comment by andreaangiolillo** [comment]
Hello 👋 
Thanks for opening the issue. Could you share your TF configuration so that we can replicate the issue? Keep in mind that we still do not support TAG for the cluster resource (label is different that TAG).

Thanks

**Comment by OlehHarmash** [comment]
```
resource "mongodbatlas_cluster" "cluster" {
  count        = var.create_advanced_cluster ? 0 : 1
  project_id   = var.project_id #mongodbatlas_project.project.id
  name         = var.cluster_name
  cluster_type = var.cluster_type

  replication_specs {
    num_shards = var.cluster_num_shards

    dynamic "regions_config" {
      for_each = var.regions_config
      content {
        region_name     = regions_config.value.region_name
        electable_nodes = regions_config.value.electable_nodes
        priority        = regions_config.value.priority
        read_only_nodes = regions_config.value.read_only_nodes
      }
    }
  }

  cloud_backup                 = var.cluster_cloud_backup
  auto_scaling_disk_gb_enabled = var.cluster_auto_scaling_disk_gb_enabled
  #Error: error creating MongoDB Cluster: POST 
  #https://cloud.mongodb.com/api/atlas/v1.0/groups/63efaf33be7e9d6c15281ee3/clusters: 400 (request 
  #"ATLAS_CLUSTER_VERSION_DEPRECATED") MongoDB version is deprecated in Atlas.
  #mongo_db_major_version       = var.cluster_mongo_db_major_version

  # Provider Settings "block"
  provider_name               = "AZURE"
  provider_disk_type_name     = var.cluster_provider_disk_type_name
  provider_instance_size_name = var.cluster_provider_instance_size_name

  dynamic "labels" {
    for_each = var.cluster_labels
    content {
      key   = labels.value.key
      value = labels.value.value
    }
  }
}
```

IT is a part of module.
Thank you! I got it. I thought that label == TAG.

**Comment by andreaangiolillo** [comment]
Hi @OlehHarmash 
Thank you for sharing the additional information. Unfortunately, the provider does not have support for TAG. However, our plan is to include this functionality over the next few months. Feel free to reach out if you need anything else here in the interim.

Thanks

**Comment by andyfisher100** [comment]
@andreaangiolillo are you able to explain what the "labels" section in the documentation refers to then? 

Like @OlehHarmash stated, I assumed "Labels" from the documentation was the same as "tags" in the atlas UI but I have tested this today and it does not work as stated in this bug

[Provider Docs](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cluster#labels)

**Comment by andreaangiolillo** [comment]
Hi @andyfisher100 👋 
According to the [api doc: createCluster](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Clusters/operation/createCluster):


> labels: Collection of key-value pairs between 1 to 255 characters in length that tag and categorize the cluster. The MongoDB Cloud console doesn't display your labels. Cluster labels will be deprecated in a future release. We strongly recommend that you use [resource tags](https://dochub.mongodb.org/core/add-cluster-tag-atlas) instead

I will open a PR to update our provider documentation to make this more clear. Thanks for the flag.

**Comment by andyfisher100** [comment]
Perfect, thank you! I didn't realise that labels were different to tags, I assumed the terminology was just being used interchangeably.

Is there any better timescale for tags? I know you mentioned over the next few months but just wondered if there was anything more specific? This is something we need as soon as possible to help us better understand the clusters deployed in our organisation

**Comment by andreaangiolillo** [comment]
> Is there any better timescale for tags? I know you mentioned over the next few months but just wondered if there was anything more specific? This is something we need as soon as possible to help us better understand the clusters deployed in our organisation

Unfortunately, I don't have more details. Tagging our Product Manager @Zuhairahmed to see if he can share more details regarding this feature. Thanks

**Comment by andreaangiolillo** [comment]
Hello @andyfisher100 @OlehHarmash,

We recently released v1.12.0 which support cluster tag. (see [CHANGELOG.md](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CHANGELOG.md#v1120-2023-09-20)). Feel free to open a new issue if you are not able to use this new feature in v1.12.0. Thanks!
